File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            int counter = 0;
            for (int j = 0; j < nums.length; j++) {
                if (nums[i] == nums[j]) {
                    counter++;
                }
            }
            if (counter > 1) {
                return true; 
            }
        }
        return false; 
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        product = 1
        zero_amount = 0 
        output = [0] * len(nums)
        for i in range(len(nums)):
            if nums[i] == 0 :
                zero_amount += 1
            else: 
                product *= nums[i]
        if zero_amount > 1 :
            return [0] * len(nums)
        
        for i in range(len(nums)):
            if zero_amount != 0: 
                output[i]
                if nums[i] != 0:
                    output[i] = 0
                else:
                    output[i] = product
            else:
                output[i] = product // nums[i] 
        return output


                
                    